satellite/marketingweb/form.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package marketingweb
+
+import (
+	"net/http"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/schema"
+
+	"storj.io/storj/satellite/rewards"
+)
+
+// parseOfferForm decodes POST form data into a new offer.
+func parseOfferForm(w http.ResponseWriter, req *http.Request) (o rewards.NewOffer, e error) {
+	err := req.ParseForm()
+	if err != nil {
+		return o, err
+	}
+
+	if err := decoder.Decode(&o, req.PostForm); err != nil {
+		return o, err
+	}
+
+	return o, nil
+}
+
+var (
+	decoder = schema.NewDecoder()
+)
+
+// init safely registers convertStringToTime for the decoder.
+func init() {
+	decoder.RegisterConverter(time.Time{}, convertStringToTime)
+	decoder.RegisterConverter(rewards.USD{}, convertDollarsToUSD)
+}
+
+// convertDollarsToUSD formats form time input as time.Time.
+func convertDollarsToUSD(s string) reflect.Value {
```

Comment:
This name feels confusing for me since `Dollars` is already in `USD`

Reply:
this is converting dollar int to a usd type... and yeah, this similarity is unfortunate